python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -319,13 +340,40 @@ def get_materialized_partitions(
             if count > 0
         ]
 
+    def get_dynamic_partitions(self, partitions_def_name: str) -> Sequence[str]:
+        """Returns a list of partitions for a partitions definition."""
+        if partitions_def_name not in self._dynamic_partitions_cache:
+            self._dynamic_partitions_cache[
+                partitions_def_name
+            ] = self.instance.get_dynamic_partitions(partitions_def_name)
+        return self._dynamic_partitions_cache[partitions_def_name]
+
+    ####################
+    # RECONCILIATION
+    ####################
+
+    def get_latest_storage_id(self, event_type: DagsterEventType) -> Optional[int]:
```

Comment:
reasonable question -- considering this function is not actually caching anything and is only used once (in asset backfill land), it's debatable if it seems like it might be more appropriate to move this out of the class entirely.